### PR TITLE
feat(unlock-app) - custom credit card price fe 

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

This PR enables custom credit card price for a specific lock.

The custom credit card price can be set from the settings page

## Custom price section
- A toggle is added to enable/disable the custom price 
- This section is visible only for lock managers when the credit card payment is enabled 

Screenshot 2023-06-15 at 22 18 01

## Set custom price 

- When `use fixed price` is enabled we can set a custom price 

Screenshot 2023-06-15 at 17 36 48

- Validation is in place to make sure that we are settings the correct values (also the min allowed value for credit card price

Screenshot 2023-06-15 at 22 26 56

- When conversion is not possible  for a specific currency we show an alert 

Screenshot 2023-06-15 at 22 46 22

## Custom price in checkout 

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  }
+
+  const saveDisabled = isSaveLockSettingLoading || disabled
+
+  const symbol = lock?.currencySymbol
+
+  return (
+    <div className="grid gap-2">
+      <SettingCardDetail
+        title="Set fixed price in $USD"
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx:111]:
```suggestion
        title="Set fixed price in USD"
```

File: unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            />
+          )}
+          <div className="w-full md:w-1/3">
+            <Button className="w-full" size="small" disabled={saveDisabled}>
+              Apply
+            </Button>
+          </div>
+        </form>
+      </div>
+      {!hasPriceConversion && (
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Settings/forms/CreditCardCustomPrice.tsx:161]: Please enable the toggle by default if indeed `hasPriceConversion` is falsy and only show the message below if the user has not set a price for cards.

**Review by julien51 — APPROVED** [review]
I like the approach!

**Comment by kalidiagne** [comment]
Sceens for the last update

**Show custom credit card in lock selection step**

Screenshot 2023-06-19 at 13 49 20

**Show custom credit card price quantity step** 

Screenshot 2023-06-19 at 13 52 26

**Comment by kalidiagne** [comment]
**Show gas cost to match the USD total when custom credit card price is present** 


Screenshot 2023-06-19 at 14 29 43

**Review by searchableguy — APPROVED** [review]

**Pull request merged**